src/app/services/search.service.ts

```diff
@@ -1,40 +1,26 @@
 import { Injectable } from '@angular/core';
-import { HttpClient, HttpParams } from '@angular/common/http';
+import { HttpClient } from '@angular/common/http';
 import { Observable, throwError } from 'rxjs';
-import { catchError, map, retry } from 'rxjs/operators';
+import { catchError, retry } from 'rxjs/operators';
 
 import { SearchServiceConfig } from '.';
 import { ApiResponse } from '../models/api-response';
 
 @Injectable()
 export class SearchService {
-	private static readonly apiUrl: URL = new URL('https://api.loklak.org/api/search.json');
 
 	constructor(
 		private http: HttpClient
 	) { }
 
 	public fetchQuery(query: string, config: SearchServiceConfig): Observable<ApiResponse> {
-		const searchParams = new HttpParams();
-		searchParams.set('q', query);
-		searchParams.set('callback', 'JSONP_CALLBACK');
-		searchParams.set('minified', 'true');
-		searchParams.set('source', config.source);
-		searchParams.set('maximumRecords', config.maximumRecords.toString());
-		searchParams.set('timezoneOffset', config.getTimezoneOffset());
-		searchParams.set('startRecord', config.startRecord.toString());
-
-		if (config.getAggregationFieldString()) {
-			searchParams.set('fields', config.getAggregationFieldString());
-			searchParams.set('limit', config.aggregationLimit.toString());
-		}
-
-		if (config.getFilterString()) {
-			searchParams.set('filter', config.getFilterString());
-		}
-
-		const jsonpUrl = SearchService.apiUrl.toString() + `?${searchParams.toString()}`;
-
+		const jsonpUrl = 'https://api.loklak.org/api/search.json' +
+							'?q=' + query +
```

Comment:
This was working fine for Angular 4. But it is working unexpectedly for Angular 6. I am looking for its alternative solution. It could be updated in further patches.